toonz/sources/tnztools/brushtool.cpp

```diff
@@ -1292,94 +1292,102 @@ void BrushTool::leftButtonDrag(const TPointD &pos, const TMouseEvent &e) {
 
 //---------------------------------------------------------------------------------------------------------------
 
-void BrushTool::leftButtonUp(const TPointD &pos, const TMouseEvent &e) {
-  bool isValid = m_enabled && m_active;
-  m_enabled    = false;
 
-  if (!isValid) return;
+void BrushTool::doPathStuff(TVectorImageP vi) {
+	double error = 20.0 * getPixelSize();
+	flushTrackPoint();
+	TStroke *stroke = m_track.makeStroke(error);
 
-  if (m_isPath) {
-    double error = 20.0 * getPixelSize();
-    flushTrackPoint();
-    TStroke *stroke = m_track.makeStroke(error);
-    int points      = stroke->getControlPointCount();
+	if (!isJustCreatedSpline(vi.getPointer())) {
+		m_isPrompting = true;
 
-    if (TVectorImageP vi = getImage(true)) {
-      struct Cleanup {
-        BrushTool *m_this;
-        ~Cleanup() { m_this->m_track.clear(), m_this->invalidate(); }
-      } cleanup = {this};
+		QString question("Are you sure you want to replace the motion path?");
+		int ret =
+			DVGui::MsgBox(question, QObject::tr("Yes"), QObject::tr("No"), 0);
 
-      if (!isJustCreatedSpline(vi.getPointer())) {
-        m_isPrompting = true;
+		m_isPrompting = false;
 
-        QString question("Are you sure you want to replace the motion path?");
-        int ret =
-            DVGui::MsgBox(question, QObject::tr("Yes"), QObject::tr("No"), 0);
+		if (ret == 2 || ret == 0) return;
```

Comment:
(continued from the previous comment)
So, I think you need to call `m_track.clear()` and `invalidate()` here if you do not use `Cleanup` .